data-prepper-plugins/rds-source/src/test/java/org/opensearch/dataprepper/plugins/source/rds/stream/StreamWorkerTaskRefresherTest.java

```diff
@@ -106,135 +113,270 @@ class StreamWorkerTaskRefresherTest {
     @Mock
     private BinlogEventListener binlogEventListener;
 
+    @Mock
+    private LogicalReplicationEventProcessor logicalReplicationEventProcessor;
+
     @Mock
     private DbTableMetadata dbTableMetadata;
 
     @Mock
     private GlobalState globalState;
 
-    private StreamWorkerTaskRefresher streamWorkerTaskRefresher;
+    @Mock
+    private MySqlSchemaManager mySqlSchemaManager;
 
-    @BeforeEach
-    void setUp() {
-        when(pluginMetrics.counter(CREDENTIALS_CHANGED)).thenReturn(credentialsChangeCounter);
-        when(pluginMetrics.counter(TASK_REFRESH_ERRORS)).thenReturn(taskRefreshErrorsCounter);
-        when(executorServiceSupplier.get()).thenReturn(executorService).thenReturn(newExecutorService);
-        when(sourceConfig.getEngine()).thenReturn(EngineType.MYSQL);
-        streamWorkerTaskRefresher = createObjectUnderTest();
-    }
+    @Mock
+    private PostgresSchemaManager postgresSchemaManager;
+
+    private StreamWorkerTaskRefresher streamWorkerTaskRefresher;
 
-    @Test
-    void test_initialize_then_process_stream() {
-        when(replicationLogClientFactory.create(streamPartition)).thenReturn(binaryLogClientWrapper);
-        when(binaryLogClientWrapper.getBinlogClient()).thenReturn(binaryLogClient);
-        final Map<String, Object> progressState = mockGlobalStateAndProgressState();
-        try (MockedStatic<StreamWorker> streamWorkerMockedStatic = mockStatic(StreamWorker.class);
-             MockedStatic<BinlogEventListener> binlogEventListenerMockedStatic = mockStatic(BinlogEventListener.class);
-             MockedStatic<DbTableMetadata> dbTableMetadataMockedStatic = mockStatic(DbTableMetadata.class)) {
-            dbTableMetadataMockedStatic.when(() -> DbTableMetadata.fromMap(progressState)).thenReturn(dbTableMetadata);
-            streamWorkerMockedStatic.when(() -> StreamWorker.create(eq(sourceCoordinator), any(ReplicationLogClient.class), eq(pluginMetrics)))
-                    .thenReturn(streamWorker);
-            binlogEventListenerMockedStatic.when(() -> BinlogEventListener.create(eq(streamPartition), eq(buffer), any(RdsSourceConfig.class),
-                            any(String.class), eq(pluginMetrics), eq(binaryLogClient), eq(streamCheckpointer),
-                            eq(acknowledgementSetManager), eq(dbTableMetadata), any(CascadingActionDetector.class)))
-                    .thenReturn(binlogEventListener);
-            streamWorkerTaskRefresher.initialize(sourceConfig);
+    @Nested
```

Comment:
This test class was refactored to have two nested test classes, one for MySQL (remain unchanged), the other for Postgres (newly added).